File: guides/micronaut-security-keys-jwks/kotlin/src/main/kotlin/example/micronaut/JwkConfigurationProperties.kt
package example.micronaut

import io.micronaut.context.annotation.ConfigurationProperties
import jakarta.validation.constraints.NotBlank

@ConfigurationProperties("jwk") // <1>
class JwkConfigurationProperties : JwkConfiguration {

    @NotBlank // <2>
    override lateinit var primary: String

    @NotBlank // <2>
    override lateinit var secondary: String
}

File: guides/micronaut-security-keys-jwks/python/src/example/micronaut/jwk_configuration_properties.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import NotBlank
from micronaut.context.annotation import ConfigurationProperties

from .jwk_configuration import JwkConfiguration


# tag::clazz[]
@ConfigurationProperties("jwk")  # <1>
@dataclass
class JwkConfigurationProperties(JwkConfiguration):
    primary: Annotated[str, NotBlank]  # <2>
    secondary: Annotated[str, NotBlank]  # <2>
# end::clazz[]
